llvm/lib/Target/AMDGPU/SIInstrInfo.cpp

```diff
@@ -6231,10 +6231,11 @@ void SIInstrInfo::legalizeGenericOperand(MachineBasicBlock &InsertMBB,
     return;
 
   Register DstReg = MRI.createVirtualRegister(DstRC);
+  Op.setSubReg(0);
```

Comment:
Instead of clearing this and using .add, could use .addReg and just not take the subreg index 

We probably don't want to be preserving the flags here. If this somehow ended up with a kill, it would be wrong